src/structstore_pybind.cpp

```diff
@@ -243,12 +260,33 @@ void register_structstore_pybind(py::module_& m) {
         FieldAccess access = list.insert(index);
         from_object(access, value, std::to_string(index));
     });
+    list.def("extend", [](List& list, py::handle& value) {
+        auto lock = list.write_lock();
+        for (const auto& val : value.cast<py::list>()) {
+            from_object(list.push_back(), val, std::to_string(list.size() - 1));
```

Comment:
```suggestion
            std::string field_name = std::to_string(list.size());
            from_object(list.push_back(), val, field_name);
```
Minor nitpick, but I guess this makes it more robust against the argument evaluation order, same applies to the other occurrences below